### PR TITLE
Added generic logging interface

### PR DESCRIPTION
**Pull request by dogversioning**
This PR moves logging infrastructure from an ad hoc 'as you need it' kind of interface to something in a dedicated module.

### Checklist
- [X] Consider if documentation (like in `docs/`) needs to be updated
- [X] Consider if tests should be added
- [X] Update template repo if there are changes to study configuration

### CODE REVIEW

File: cumulus_library/actions/file_generator.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def run_generate_markdown(
         if len(count_tables) > 0:
             f.write(f"## {manifest_parser.get_study_prefix()} count tables\n\n")
             for table in count_tables:
-                manifest_parser._write_md_table(table, study_df, f)
+                _write_md_table(table, study_df, f)
```
Comment (dogversioning) [inline cumulus_library/actions/file_generator.py:88]: this is an unrelated bug in the markdown generator I found this morning

File: cumulus_library/protected_table_builder.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
 """Builder for creating tables for tracking state/logging changes"""
 
-from cumulus_library import base_table_builder, enums, study_parser
+from cumulus_library import base_table_builder, databases, enums, study_parser
 from cumulus_library.template_sql import base_templates
 
-TRANSACTIONS_COLS = ["study_name", "library_version", "status", "event_time"]
-TRANSACTION_COLS_TYPES = ["varchar", "varchar", "varchar", "timestamp"]
+TRANSACTIONS_COLS = ["study_name", "library_version", "status", "event_time", "message"]
+TRANSACTION_COLS_TYPES = ["varchar", "varchar", "varchar", "timestamp", "varchar"]
```
Comment (dogversioning) [inline cumulus_library/protected_table_builder.py:7]: i talked about adding a gentle upgrade path for this, but if we're making this part of a 3.0 breaking changeset, i'm inclined to say something like 'just remove previous lib_transaction tables'. Can be talked out of this position.

File: cumulus_library/template_sql/insert_into.sql.jinja [hunk 3]
```diff
@@ -12,6 +12,8 @@ VALUES
     {%- for field in row -%}
     {%- if table_cols[loop.index0] in type_casts.keys() -%}
     {{ type_casts[table_cols[loop.index0]] }} '{{ field }}'
+    {%- elif not field -%}
+    NULL
```
Comment (dogversioning) [inline cumulus_library/template_sql/insert_into.sql.jinja:16]: not actually a bug per se, just the first time we actually tried to explicitly insert null data via this template.

Comment (mikix) [reply cumulus_library/template_sql/insert_into.sql.jinja:16]: I've hit errors with Athena where it likes that kind of thing as `CAST(NULL AS VARCHAR)` - are you gonna need that here too?

File: cumulus_library/log_utils.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        cursor=cursor,
+        schema=schema,
+        manifest=manifest,
+        dataset=[
+            [
+                manifest.get_study_prefix(),
+                __version__,
+                status.value,
+                base_utils.get_utc_datetime(),
+                message if message else None,
```
Comment (mikix) [inline cumulus_library/log_utils.py:41]: nit: I think `message or None` is maybe slightly more readable, but if/else is good too.

**Review by mikix — APPROVED** [review]

File: cumulus_library/template_sql/insert_into.sql.jinja (hunk 3 above)
Comment (dogversioning) [reply cumulus_library/template_sql/insert_into.sql.jinja:16]: almost certaintly, yeah
